packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -1191,7 +1235,7 @@ class RenderEditable extends RenderBox {
   /// When [ignorePointer] is true, an ancestor widget must respond to tap
   /// down events by calling this method.
   void handleTapDown(TapDownDetails details) {
-    _lastTapDownPosition = details.globalPosition - _paintOffset;
+    _lastTapDownPosition = details.globalPosition;
```

Comment:
I'm surprised that this change doesn't have a corresponding number of other changes doing the opposite. What's the story here?

Reply:
I thought it was not tested because the selection when scrolling was completely broken, but when I dug further I discovered that it was broken in commit 892c8919c7, where the correct behavior was implemented in `selectWordEdge()`, which is used on iOS, but broken for Android, which uses `selectPositionAt()` and `selectWordsInRange()`.

The `_paintOffset` is already subtracted in `selectPositionAt()` and `selectWordsInRange()`, but was not subtracted in `selectWordEdge()`. So I've updated `selectWordEdge()`. All the uses now correct subtract `_paintOffset` exactly once.

Reply:
Can you add some specific test cases that cover this (outside of their use in the selection handle test cases) to ensure that we never break this again?

Reply:
Working on this. I wrote a test which uncovered a strange issue with taps not being received when they should be.

Reply:
It looks the original issue I was solving here (#27948) was already fixed in (#28242).
The "issue" I ran into in the test was just a "user-error" caused by the xster's refactoring of the gesture logic.